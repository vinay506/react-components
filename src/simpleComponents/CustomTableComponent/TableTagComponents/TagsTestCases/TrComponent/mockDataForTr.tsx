import {TextWithTooltip} from '../../../../../compoundComponents';
import {InfoIcon} from '../../../../index';
import React from 'react';

const headerColumns = [{title:"Tests",styles:{fontWeight:'bold',height:'45px'}}];
const bodyColumns = [{key:"column1",styles:{fontWeight:'bold',height:'45px'}},{key:"column2",styles:{height:'45px'}},
{key:"column3",styles:{height:'45px'}},{key:"column4",styles:{height:'45px'}}];

const record = [{
    column1:  <TextWithTooltip text='Chapter Test' tooltip={{id:'chapterTest',backgroundColor:'#d8c7c7',textColor:'#000',multiline:true,place:'bottom'}} tooltipText="Chapter Test" ><InfoIcon styles={{width:'20px',height:'15px'}}></InfoIcon></TextWithTooltip>,
    column2: 'Selected Chapters (30)',
    column3: 'All Chapters (91)',
    column4: 'All Chapters (91)'
  }]

export {headerColumns,bodyColumns,record}