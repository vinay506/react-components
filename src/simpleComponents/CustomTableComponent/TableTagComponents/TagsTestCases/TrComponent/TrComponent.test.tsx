import React from 'react';
import { shallow } from 'enzyme';
import { CustomTableRow, CustomTdCell, CustomThCell } from '../../index';
import renderer from 'react-test-renderer'
import { bodyColumns, headerColumns,record } from './mockDataForTr'

function getClasses(props: any) {
  let classes = 'rmwc-data-table__row';
  classes = (props.headerRow) ? classes + ' mdc-data-table__header-row' : classes + ' mdc-data-table__row';
  return classes
}

const getCustomTableCells = (row: any) => {
  return (bodyColumns || []).map((column: any, index: number) => {
    return <CustomTdCell style={column.styles} key={'bdcol-' + index}>{row[column.key]}</CustomTdCell>
  })
}

const getCustomTableHeaderCells = () => {
  return (headerColumns || []).map((column: any, index: number) => {
    return <CustomThCell style={column.styles} key={'hdcol-' + index}>{column.title}</CustomThCell>
  })
}


describe('<Custom TR  >', function () {
  let wrapper: any;
  let props: any;
  let tree: any;
  beforeEach(() => {
    props = {}
  });

  // Snapshot test
  it('should render', () => {
    tree = renderer.create(<CustomTableRow {...props}></CustomTableRow>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render Tr', () => {
    wrapper = shallow(<CustomTableRow {...props}></CustomTableRow>)
    const tr = wrapper.find('tr');
    expect(tr.exists()).toBe(true);
  })

})

describe('<Custom TR with header cell', function () {
  let props: any;
  let wrapper: any;
  beforeEach(() => {
    props = {}
    wrapper = shallow(<CustomTableRow headerRow {...props}> {getCustomTableHeaderCells()}</CustomTableRow>)
  });

  // Snapshot test
  it('should render custom rable row with custom header ', () => {
    const tree = renderer.create(<CustomTableRow headerRow {...props}> {getCustomTableHeaderCells()}</CustomTableRow>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render CustomTh cells', () => {
    const tr = wrapper.find('tr');
    const thcells = tr.find(CustomThCell);

    expect(thcells.exists()).toBe(true);
  })

  it('Counts the no of  customThcells', () => {
    const tr = wrapper.find('tr');
    const thcells = tr.find(CustomThCell);
    expect(thcells).toHaveLength(1)
    
  }) 
});


describe('<Custom TR with body cell', function () {
  let props: any;
  let wrapper: any;
  beforeEach(() => {
    props = {}
    wrapper = shallow(<CustomTableRow headerRow {...props}> {getCustomTableCells(record)}</CustomTableRow>)
  });

  // Snapshot test
  it('should render custom rable row with custom td cells', () => {
    const tree = renderer.create(<CustomTableRow headerRow {...props}> {getCustomTableCells(record)}</CustomTableRow>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render CustomTd cells', () => {
    const tr = wrapper.find('tr');
    const tdcells = tr.find(CustomTdCell);
    expect(tdcells.exists()).toBe(true);
  })

  it('Counts the no of customTdcells', () => {
    const tr = wrapper.find('tr');
    const tdcells = tr.find(CustomTdCell);
    expect(tdcells).toHaveLength(4);    
  }) 
});