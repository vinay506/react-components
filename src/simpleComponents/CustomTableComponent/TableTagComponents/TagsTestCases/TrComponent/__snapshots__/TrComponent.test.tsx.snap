// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Custom TR  > should render 1`] = `
<tr
  className="rmwc-data-table__row mdc-data-table__row"
/>
`;

exports[`<Custom TR with body cell should render custom rable row with custom td cells 1`] = `
<tr
  className="rmwc-data-table__row mdc-data-table__header-row"
>
   
  <td
    className="mdc-data-table__cell rmwc-data-table__cell"
    style={
      Object {
        "fontWeight": "bold",
        "height": "45px",
      }
    }
  >
    X
  </td>
  <td
    className="mdc-data-table__cell rmwc-data-table__cell"
    style={
      Object {
        "height": "45px",
      }
    }
  >
    X
  </td>
  <td
    className="mdc-data-table__cell rmwc-data-table__cell"
    style={
      Object {
        "height": "45px",
      }
    }
  >
    X
  </td>
  <td
    className="mdc-data-table__cell rmwc-data-table__cell"
    style={
      Object {
        "height": "45px",
      }
    }
  >
    X
  </td>
</tr>
`;

exports[`<Custom TR with header cell should render custom rable row with custom header  1`] = `
<tr
  className="rmwc-data-table__row mdc-data-table__header-row"
>
   
  <th
    className="rmwc-data-table__cell mdc-data-table__header-cell"
    style={
      Object {
        "fontWeight": "bold",
        "height": "45px",
      }
    }
  >
    Tests
  </th>
</tr>
`;
